app/channel.go

```diff
@@ -2541,6 +2541,15 @@ func (a *App) MarkChannelAsUnreadFromPost(postID string, userID string, collapse
 				message.Add("thread", payload)
 				a.Publish(message)
 			}
+		} else if !threadMembership.Following && followThread {
+			opts := store.ThreadMembershipOpts{
+				Following:       true,
+				UpdateFollowing: true,
+			}
+			_, storeErr = a.Srv().Store.Thread().MaintainMembership(user.Id, threadId, opts)
+			if storeErr != nil && !errors.As(storeErr, &nfErr) {
+				return nil, model.NewAppError("MarkChannelAsUnreadFromPost", "app.channel.update_last_viewed_at_post.app_error", nil, storeErr.Error(), http.StatusInternalServerError)
```

Comment:
Slightly unrelated but the error id used in this method doesn't look right. Maybe it got moved around.

Reply:
You're right, id doesn't look right but the text it translate to works for this scenario so it's probably ok. Looking forward to us removing a lot of these translations